---
title: "MTseeker example"
shorttitle: "MTseeker guide"
author: 
  - Tim Triche, Jr. 
  - Noor Sohail 
package: MTseeker
abstract: >
  An example of processing renal oncocytomas and matched normal
  kidney samples with MTseeker, calling variants and plotting results.
vignette: >
  %\VignetteIndexEntry{MTseeker example}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output: 
  BiocStyle::html_document:
    toc_float: true
---

# Introduction

MTseeker works best when given some interesting mitochondrial data to work with.
Renal oncocytomas are great big pink cells that are jammed full of defective 
mitochondria, and sometimes progress to genomically unstable chromophobe renal
cell carcinomas (kidney cancer) in unlucky hosts. Nobody seems to be entirely 
sure what role mitochondrial variants play in their evolution, but the cells 
have thousands of mitochondria stuffed into them. So that's what we'll study.


# Loading data

First we needed to load the oncocytoma BAMs. We don't actually do this here, 
since they are several gigabytes apiece, but notice that all of them have 
been aligned with BWA against the canonical rCRS mitogenome by splicing it 
into hg19. (As opposed to GRCh37, which is what we should have done... but
the point is that any modern GRCh assembly *or* a spliced rCRS contig works.)

```{r loadBams, eval=FALSE} 
library(MTseeker)

if (FALSE) { 
  # we use SamBlaster... a lot... in my lab.
  BAMfiles <- grep("(split|disc)", value=T, invert=T, list.files(patt=".bam$"))
  names(BAMfiles) <- sapply(strsplit(BAMfiles, "\\."), `[`, 1)
  BAMs <- data.frame(BAM=BAMfiles, 
                     Sample_Group=ifelse(grepl("NKS", BAMfiles), 
                                         "normal","tumor"))
  rownames(BAMs) <- sub("NKS", "normal", sub("RO","oncocytoma", rownames(BAMs)))
  BAMs$subject <- as.integer(sapply(strsplit(BAMs$BAM, "(_|\\.)"), `[`, 2))

  # we merged all the BAMs after-the-fact, so...
  BAMs <- subset(BAMs, grepl("merged", BAMs$BAM))
  BAMs <- BAMs[order(BAMs$subject), ]

  library(parallel) 
  options("mc.cores"=detectCores())
  MTreads <- getMT(BAMs, filter=FALSE) 
  names(MTreads) <- sapply(strsplit(fileName(MTreads), "\\."), `[`, 1)
  saveRDS(MTreads, file="oncocytoma_and_matched_normal_MTreads.rds")
}
```

Since realigning 22 whole exomes and extracting/counting reads takes a while,
we created the MTseekerData package to hold the output from doing the above. 
The _RONKSreads_ and _RONKSvariants_ data objects hold what you might expect 
them to hold, at least once you find out that RONKS stands for _R_enal 
_O_ncocytoma and _N_ormal _K_idney _S_amples from the same patients. 


# Relative mitochondrial copy number changes

We'd like to compute the relative mitochondrial copy number for each:

```{r computeCN}
library(MTseekerData)
data(RONKSreads, package="MTseekerData")
mVn <- Summary(RONKSreads)$mitoVsNuclear
names(mVn) <- names(RONKSreads) 
CN <- mVn[seq(2,22,2)]/mVn[seq(1,21,2)] 
mtCN <- data.frame(subject=names(CN), CN=CN)

library(ggplot2) 
library(ggthemes)
p <- ggplot(mtCN, aes(x=subject, y=CN, fill=subject)) + 
       geom_col() + theme_tufte() + ylim(0,5) + 
       ylab("Tumor/normal mitochondrial ratio") + 
       ggtitle("Mitochondrial retention in oncocytomas")
print(p)
```

# Calling variants

Obviously it's not much good to have a variant caller that can't call variants,
so we demonstrate that here. (Note: tumor/normal calls, haplogroup inference,
and soft-backfiltering of haplogroup-determining variants are works in progress,
so we do not currently demonstrate them here, although the fpFilter datasets 
are useful for these purposes)

gmapR can be a bit fiesty, so we simply document the process below: 

```{r callVariants, eval=FALSE} 
if (FALSE) { 
  RONKSvariants <- callMT(RONKSreads)
  save(RONKSvariants, file="RONKSvariants.rda")
}
```

For this vignette, we have stored the results in the MTseekerData package:

```{r loadVariants}
library(MTseekerData)
data(RONKSvariants, package="MTseekerData")
```

# Plotting variants

Show off the results:

```{r plotVariants} 
plot(filt(subset(RONKSvariants)))
```

The mitochondrial variant plots are structured sort of like tree rings, in that
each sample is a "ring" and each point along the circle is annotated against
a region of the mitochondrial genome. The filt() method drops variants that do
not pass remedial quality filters, but it is advisable to also filter on VAF 
(0.03, or 3%, is a recommended floor to keep out NuMT nuclear-mitochondrial 
integrations; see Ju et al, eLife 2014, for more on this subject) and mask off
regions of frequent homopolymer miscalls, and also haplogroup-related variants.


# Plotting functional impacts

Now let's plot a cartoon of the putative functional impact in one patient (RO1):

```{r makeSVG}
data(RONKSvariants, package="MTseekerData")
SVG <- mtComplex(RONKSvariants[[2]]) 
```

The above will bring up an image in a browser window like the one in the README.
You can also generate a PDF file of the modified rendering if you prefer:

```{r makePDF, eval=FALSE}
library(rsvg) 
tmppdf <- paste(tempdir(), "RO_1.functionalAnnot.pdf", sep="/") 
rsvg_pdf(tmppdf)
```

We would like to add a biohazard/mutagen symbol to complexes within the electron
transport chain (ETC) that are impacted by nonsynonymous variants, and this is 
in progress; the output is Scalable Vector Graphics (SVG) based on an image 
created and shared by Tim Vickers at Washington University in St. Louis. Any
suggestions regarding how to scale this visual up to populations of cells, 
people, or organisms are welcome; some components (such as in Complex II) have 
migrated to the nuclear genome in humans, while others are retained in mtDNA in 
humans but lost to nuclear genomes in other eukaryotes. Moreover, tendencies 
for particular diseases or conditions to hit particular complexes are of both
biological and medical interest, which is why we added this in the first place.

We hope you enjoy working with mitochondrial genomes as much as we have. 
