<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Bioconductor tools for mitochondrial variant calling • MTseeker</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Bioconductor tools for mitochondrial variant calling">
<meta property="og:description" content="Variant analysis tools for mitochondrial genetics. ">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">MTseeker</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.9.47</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
    
<p><img align="left" width="500" height="478" src="inst/extdata/RO_NKS_filtered.png"></p>
<div id="mtseeker" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#mtseeker" class="anchor"></a>MTseeker</h1></div>
<p>MTseeker is a toolkit for mitochondrial variant analysis in the <a href="http://bioconductor.org/">Bioconductor</a> environment. Given suitably aligned reads, it will call variants (and/or mitochondrial copy number), plot them for multiple samples, and provide basic functional annotation for coding variants. If we were cool people, it would have a mitochondrion stuffed into a hexagon, but we don’t, so it doesn’t… <em>yet</em>.</p>
<p><a href="https://travis-ci.org/trichelab/MTseeker/MTseeker"><img src="https://travis-ci.org/trichelab/MTseeker/MTseeker.png?branch=master" alt="Build Status"></a></p>
<div id="how-to-install" class="section level2">
<h2 class="hasAnchor">
<a href="#how-to-install" class="anchor"></a>How to install</h2>
<pre><code><a href="http://www.rdocumentation.org/packages/devtools/topics/install_github">devtools::install_github("trichelab/MTseeker")
# or:
install.packages("BiocManager")
BiocManager::install("trichelab/MTseeker")</a></code></pre>
</div>
<div id="how-it-works" class="section level2">
<h2 class="hasAnchor">
<a href="#how-it-works" class="anchor"></a>How it works</h2>
<p>Visit the vignettes for more details, but suppose we have a bunch of renal oncocytomas (known to be driven early in development by mitochondrial mutations) and matched adjacent normal kidney samples (let’s call them “RO” and “NKS”).</p>
<pre class="{r}"><code>MTreads &lt;- getMT(BAMfiles)</code></pre>
<p>The first and simplest thing we might like to do is get mitochondrial reads from exome sequencing BAMs (or, really, any BAM – it doesn’t have to be any particular sequencing technology, though at present we support human mitogenomes far better than any other organism). Above, the getMT function does that.</p>
<p>Now we might also like to estimate from the mt/nuclear read support what the rough ratio of mitochondria in the tumor tissue is compared to the normal tissue. Since we track the read support for each upon load, we can use that:</p>
<pre class="{r}"><code>mVn &lt;- Summary(MTreads)$mitoVsNuclear
names(mVn) &lt;- names(MTreads) 
CN &lt;- mVn[seq(2,22,2)]/mVn[seq(1,21,2)] 
mtCN &lt;- data.frame(subject=names(CN), CN=CN)

library(ggplot2) 
library(ggthemes)
p &lt;- ggplot(mtCN, aes(x=subject, y=CN, fill=subject)) + 
  geom_col() + theme_tufte() + ylim(0,5) + 
  ylab("Tumor/normal mitochondrial ratio") + 
  ggtitle("Mitochondrial retention in oncocytomas")
# ggsave("inst/extdata/mtCN.png") 
print(p) </code></pre>
<p><img src="inst/extdata/mtCN.png" alt="mitochondrial CN"></p>
<p>Now we probably want to call variants, and then plot them (filtered):</p>
<pre class="{r}"><code>variants &lt;- callMT(MTreads)
plot(filt(variants))</code></pre>
<p>That looks like so:</p>
<p><img src="inst/extdata/RO_NKS_filtered.png" alt="mitochondrial variants for 22 samples"></p>
<p>It would be nice to have some idea what this means for each patient:</p>
<p><img src="inst/extdata/RO_1.png" alt="mitochondrial variant fallout in patient 1"></p>
<p>More to come! But this is the basic idea.</p>
</div>
</div>

  </div>

  <div class="col-md-3" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Report a bug at <br><a href="NA">NA</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Tim Triche <br><small class="roles"> Maintainer, author </small>  </li>
<li>Ben Johnson <br><small class="roles"> Author </small>  </li>
<li>Noor Sohail <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Tim Triche, Ben Johnson, Noor Sohail.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
