<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ MTseeker</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Function reference" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MTseeker</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.9.47</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="MAlignments-class.html">MAlignments-class</a></code> </p>
        </td>
        <td><p>wraps a GAlignments with information about coverage and its target BAM file</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MAlignments-methods.html">coverage(<i>&lt;MAlignments&gt;</i>)</a></code> <code><a href="MAlignments-methods.html">runLength(<i>&lt;MAlignments&gt;</i>)</a></code> <code><a href="MAlignments-methods.html">runValue(<i>&lt;MAlignments&gt;</i>)</a></code> <code><a href="MAlignments-methods.html">Summary(<i>&lt;MAlignments&gt;</i>)</a></code> <code><a href="MAlignments-methods.html">show(<i>&lt;MAlignments&gt;</i>)</a></code> <code><a href="MAlignments-methods.html">fileName(<i>&lt;MAlignments&gt;</i>)</a></code> <code><a href="MAlignments-methods.html">scanBamHeader(<i>&lt;MAlignments&gt;</i>)</a></code> </p>
        </td>
        <td><p>MAlignments methods (centralized).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MAlignments.html">MAlignments()</a></code> </p>
        </td>
        <td><p>wrap a GAlignments for easier stats</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MAlignmentsList-class.html">MAlignmentsList-class</a></code> </p>
        </td>
        <td><p>wraps a GAlignmentsList (made up of MAlignments) for nicer viewing</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MAlignmentsList-methods.html">coverage(<i>&lt;MAlignmentsList&gt;</i>)</a></code> <code><a href="MAlignmentsList-methods.html">runLength(<i>&lt;MAlignmentsList&gt;</i>)</a></code> <code><a href="MAlignmentsList-methods.html">fileName(<i>&lt;MAlignmentsList&gt;</i>)</a></code> <code><a href="MAlignmentsList-methods.html">Summary(<i>&lt;MAlignmentsList&gt;</i>)</a></code> <code><a href="MAlignmentsList-methods.html">show(<i>&lt;MAlignmentsList&gt;</i>)</a></code> </p>
        </td>
        <td><p>MAlignmentsList methods (centralized).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MAlignmentsList.html">MAlignmentsList()</a></code> </p>
        </td>
        <td><p>wrap a GAlignmentsList for viewing</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MVRanges-class.html">MVRanges-class</a></code> </p>
        </td>
        <td><p>like a VRanges, but for mitochondria</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MVRanges-methods.html">coverage(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">type(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">genes(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">snpCall(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">pos(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">show(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">annotation(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">getAnnotations(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">encoding(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">filt(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">genome(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">locateVariants(<i>&lt;MVRanges&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">tallyVariants(<i>&lt;MVRanges&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">predictCoding(<i>&lt;MVRanges&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">summarizeVariants(<i>&lt;MVRanges&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="MVRanges-methods.html">plot(<i>&lt;MVRanges&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code> </p>
        </td>
        <td><p>MVRanges methods (centralized).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MVRanges.html">MVRanges()</a></code> </p>
        </td>
        <td><p>wrap a VRanges for mitochondrial use</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MVRangesList-class.html">MVRangesList-class</a></code> </p>
        </td>
        <td><p>like a VRangesList, but for mitochondria</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MVRangesList-methods.html">coverage(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">`counts<-`(<i>&lt;MVRangesList&gt;</i>,<i>&lt;RangedSummarizedExperiment&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">counts(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">genes(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">snpCall(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">getAnnotations(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">encoding(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">predictCoding(<i>&lt;MVRangesList&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">show(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">filt(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">granges(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">summarizeVariants(<i>&lt;MVRangesList&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">genome(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">locateVariants(<i>&lt;MVRangesList&gt;</i>,<i>&lt;missing&gt;</i>,<i>&lt;missing&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">tallyVariants(<i>&lt;MVRangesList&gt;</i>)</a></code> <code><a href="MVRangesList-methods.html">plot(<i>&lt;MVRangesList&gt;</i>,<i>&lt;ANY&gt;</i>)</a></code> </p>
        </td>
        <td><p>MVRangesList methods (centralized).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="MVRangesList.html">MVRangesList()</a></code> </p>
        </td>
        <td><p>wrap a VRangesList for mitochondrial use</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="addMtGeneLabel.html">addMtGeneLabel()</a></code> </p>
        </td>
        <td><p>Set gene names for bp ranges. Original code by Stephen Turner at UVA.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="byStrand.html">byStrand()</a></code> </p>
        </td>
        <td><p>simple helper function to split a *RangesList by the strand of its mt target</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="callMT.html">callMT()</a></code> <code><a href="callMT.html">callMtVars()</a></code> </p>
        </td>
        <td><p>call mitochondrial variants against rCRS from an MAlignments[List] object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="callTNMT.html">callTNMT()</a></code> <code><a href="callTNMT.html">callTnMtVars()</a></code> </p>
        </td>
        <td><p>call mitochondrial tumor/normal variants from an MAlignments[List] object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filterMT.html">filterMT()</a></code> </p>
        </td>
        <td><p>Filter SummarizedExperiment or DataFrame values based on its $mtCovg column</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fixMetadata.html">fixMetadata()</a></code> </p>
        </td>
        <td><p>fix metadata for an MAlignmentsList or MVRangesList, if needed and possible.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="getMT.html">getMT()</a></code> </p>
        </td>
        <td><p>grab the mitochondrial reads from a BAM and estimate their fraction</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="indexMtGenome.html">indexMtGenome()</a></code> </p>
        </td>
        <td><p>build and install GmapGenome.[organism].[mtGenome], mostly for rCRS</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="injectMtVariants.html">injectMtVariants()</a></code> </p>
        </td>
        <td><p>Inject (one or more) variants against rCRS.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mtCircos.html">mtCircos()</a></code> </p>
        </td>
        <td><p>plot a canonical human (or, in principle, any) mitochondrial genome</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mtComplex.html">mtComplex()</a></code> </p>
        </td>
        <td><p>plot the (putative) functional impact of mutations to ETP genes as SVG</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mtCoverage.html">mtCoverage()</a></code> </p>
        </td>
        <td><p>mitochondrial genome coverage for an MAlignments or MVRanges object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mtHGVS.html">mtHGVS()</a></code> </p>
        </td>
        <td><p>convert mitochondrial variant calls to HGVS format for naming</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotMT.html">plotMT()</a></code> </p>
        </td>
        <td><p>plot mitochondrial variant calls in the context of heteroplasmy</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="s4Methods.html">s4Methods()</a></code> </p>
        </td>
        <td><p>List all defined methods for an S4 class (or classes, if you must)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="validMetadata.html">validMetadata()</a></code> </p>
        </td>
        <td><p>Ensure that the metadata caches in MAlignmentsLists and MVRangesLists are OK</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-all-functions">All functions</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Tim Triche, Ben Johnson, Noor Sohail.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

